components/vocabularies/hpo/api/src/main/java/org/phenotips/vocabulary/internal/solr/HumanPhenotypeOntology.java

```diff
@@ -103,6 +118,54 @@ protected int getSolrDocsPerBatch()
         return result;
     }
 
+    /**
+     * Returns the same JSON as {@link VocabularyTerm#toJson()}, with the exception that the categories are ordered.
+     * @param term whose JSON should be returned
+     * @return JSON of the `term` with ordered categories
+     */
+    public JSON termToOrderedJson(VocabularyTerm term)
+    {
+        JSON rawJson = term.toJson();
+        JSONObject json;
+        if (!rawJson.isArray()) {
+            json = (JSONObject) rawJson;
+        } else {
+            return rawJson;
+        }
+
+        Set<VocabularyTerm> categoriesSet = term.getAncestorsAndSelf();
+        List<VocabularyTerm> sortedCategories = this.sortTermsByLevel(categoriesSet);
+        List<String> sortedCategoriesIds = new LinkedList<>();
+        for (VocabularyTerm category : sortedCategories) {
+            sortedCategoriesIds.add(category.getId());
+        }
+        json.put("term_category", sortedCategoriesIds);
+        return json;
+    }
+
+    private List<VocabularyTerm> sortTermsByLevel(Collection<VocabularyTerm> terms)
+    {
+        List<VocabularyTerm> sortedTerms = new LinkedList<>();
+        final Map<VocabularyTerm, Long> levelMap = new HashMap<>();
+        for (VocabularyTerm term : terms) {
+            levelMap.put(term, term.getDistanceTo(this.getRootTerm()));
+            sortedTerms.add(term);
+        }
+        sortedTerms.sort(new Comparator<VocabularyTerm>()
+        {
```

Comment:
Jenkins build is failing. I think because List.sort() is not implemented until Java 8. I don't actually know what our current status on java 8 is. If we can't use it then just use Collections.sort()